lib/kino/proxy.ex

```diff
@@ -0,0 +1,22 @@
+defmodule Kino.Proxy do
+  @moduledoc """
+  A kino for handling proxy requests from the host.
+
+  ## Examples
+
+      Kino.Proxy.listen(fn conn ->
+        conn
+        |> Plug.Conn.put_resp_header("content-type", "application/text;charset=utf-8")
+        |> Plug.Conn.send_resp(200, "used " <> conn.method <> " method")
+      end)
+  """
+
+  @doc """
+  Persists the function to be listened by the proxy handler.
+  """
+  @spec listen(atom(), module(), (Plug.Conn.t() -> Plug.Conn.t())) ::
+          DynamicSupervisor.on_start_child()
+  def listen(name \\ __MODULE__, mod \\ Livebook.Proxy.Handler, fun) when is_function(fun, 1) do
```

Comment:
I don't think we want those parameters.

We also most likely don't want to depend on Livebook.Proxy.Handler directly. Perhaps we add a function to Kino.Bridge called `fetch_proxy_handler_spec`, and we pass the `function` as argument?

Reply:
The least coupling would be just registering the function using `Kino.Bridge` and be done, but given that we also want to use `Kino.start_child`, asking for the spec sounds sane!